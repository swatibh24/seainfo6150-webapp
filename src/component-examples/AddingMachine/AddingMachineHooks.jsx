import React, { useState } from "react";
import PropTypes from "prop-types";

const AddingMachineHooks = (props) => {
  const [currentNum, setCurrentNum] = useState(props.startNum);

  function addOne() {
    setCurrentNum(currentNum + 1);
  }


  let whee;
  if (currentNum % 10 === 0) {
    whee = <div>Whee!</div>;
  }

  return (
    <div style={{ border: "1px solid black", margin: "2rem" }}>
      <header>{props.children}</header>
      <div>We started with {props.startNum}</div>
      <div>We're now at {currentNum}</div>
      <button onClick={addOne}>Click me to add 1</button>
      {whee}
    </div>
  );
}

AddingMachineHooks.propTypes = {
  startNum: PropTypes.number.isRequired
};

export default AddingMachineHooks
