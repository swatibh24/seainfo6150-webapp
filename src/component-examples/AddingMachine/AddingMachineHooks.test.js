import React from "react";
import AddingMachineHooks from "./AddingMachineHooks";
import { fireEvent } from "@testing-library/react";

describe("AddingMachineHooks tests", () => {
  it("renders correctly", () => {
    const { container } = render(<AddingMachineHooks startNum={0} />);
    expect(container).toMatchSnapshot();
  });

  it("renders correctly when currentNum is 10", () => {
    const { container } = render(<AddingMachineHooks startNum={10} />);
    expect(container).toMatchSnapshot();
  });

  it("renders correctly after button is clicked", () => {
    const { container, getByText } = render(
      <AddingMachineHooks startNum={0} />
    );
    const button = getByText("Click me to add 1");
    fireEvent.click(button);
    expect(container).toMatchSnapshot();
  });
});
