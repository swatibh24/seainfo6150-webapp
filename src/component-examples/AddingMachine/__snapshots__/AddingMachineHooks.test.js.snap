// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddingMachineHooks tests renders correctly 1`] = `
<div>
  <div
    style="border: 1px solid black; margin: 2rem;"
  >
    <header />
    <div>
      We started with 
      0
    </div>
    <div>
      We're now at 
      0
    </div>
    <button>
      Click me to add 1
    </button>
    <div>
      Whee!
    </div>
  </div>
</div>
`;

exports[`AddingMachineHooks tests renders correctly after button is clicked 1`] = `
<div>
  <div
    style="border: 1px solid black; margin: 2rem;"
  >
    <header />
    <div>
      We started with 
      0
    </div>
    <div>
      We're now at 
      1
    </div>
    <button>
      Click me to add 1
    </button>
  </div>
</div>
`;

exports[`AddingMachineHooks tests renders correctly when currentNum is 10 1`] = `
<div>
  <div
    style="border: 1px solid black; margin: 2rem;"
  >
    <header />
    <div>
      We started with 
      10
    </div>
    <div>
      We're now at 
      10
    </div>
    <button>
      Click me to add 1
    </button>
    <div>
      Whee!
    </div>
  </div>
</div>
`;
